plugin/vscode/performance-writer.js

```diff
@@ -0,0 +1,93 @@
+// Copyright (C) 2020  Matthew Glazar
+// See end of file for extended copyright information.
+
+const fs = require('fs');
+const path = require("path");
+const process = require("process");
+
+class PerformanceWriter {
+    
+    constructor (extensionContext) {
+        this._dataAttributes = {
+            columns:'timestamp,file-ref,file-name-or-duration,event-id'
+        };
+        this._tmpLogLines = []
+        this._fileName = null;
+        this._fileIndex = null;
+        this._extensionContext = extensionContext;
+        fs.mkdirSync(this._extensionContext.logUri.fsPath)
+        this._performanceTxtPath = path.join(this._extensionContext.logUri.fsPath, "qljsPerformanceTrace.txt");
+        this._currentSessionMap = this._extensionContext.globalState.get('qljsFileMap');
+        this._writableStream = fs.createWriteStream(this._performanceTxtPath, {encoding: 'utf8', flags: 'a'})
+    }
+
+    createHeader (){
+        this.writeToTmpArray(null,"header");
+    }
+    
+    fileNameOptimizer (){
+        this._fileName = this._fileName.split(path.sep);
+        this._fileName = this._fileName[this._fileName.length - 1];
+        if (!(this._currentSessionMap)) {
```

Comment:
Performance: Move this check into the constructor.

Reply:
The PerformanceWriter class is only activated once during activateAsync and then writing performance of a new document starts with fileNameOptimizer method of this one instance of PerformanceWriter. If we were to move the _fileName into the constructor, it would imply that there is a new PerformanceWriter every time a new document is being performance traced which is completely doable, it's just not the way it's working now.

Reply:
> If we were to move the _fileName into the constructor, it would imply that there is a new PerformanceWriter every time a new document is being performance traced which is completely doable, it's just not the way it's working now.

I wasn't thinking of doing this. I was thinking that the file path could be passed down as an argument instead of being a property on the PerformanceWriter object.

I think a single PerformanceWriter object, as currently implemented, is the right solution.

Reply:
the filepath is now an argument into the initiateLogging class method